.github/actions/git-tag/action.yml

```diff
@@ -0,0 +1,31 @@
+name: "Publish GitHub Tag"
+description: "Publish GitHub Tag"
+
+inputs:
+  github_token:
+    description: 'Github token'
+    required: true
+    type: string
+  dry-run:
+    description: dry run flag
+    required: true
+    type: boolean
+
+runs:
+  using: "composite"
+  steps:
+    - name: publish tag
+      shell: bash
+      run: |
+        if [[  $BRANCH_NAME == "master" ]]; then
```

Comment:
does it resolve this variable correctly?

Reply:
I believe it does, before this step `setup` action is called which uses another action to get the `BRANCH_NAME` and save it as env variable

Reply:
best would be to declare a new input but should work also as it is now

Reply:
Added as action input